### PR TITLE
Iain8/rules update

### PR DESCRIPTION
**Pull request by iain8**
opinionated PR! open to add or remove things, but I feel we can start to get more strict with our rules and bring up our code quality

* updates parserOptions to use more recent settings and things we end up adding in every project
* alphabetises rules (because OCD)
* adds rules:
  * `curly` - we generally seem agreed to avoid single line `if`s and the like
  * `global-require` - modules are loaded synchronously so this should be done at the top of files (with rare exception e.g. newrelic), also makes it easier to see dependency usage
  * `no-dupe-keys` - easy bug reducer
  * `no-multiple-empty-lines` - less spacious
  * `no-use-before-define` - bug reducer
  * `no-useless-escape` - disabled this as it for some reason always picks up regexs
  * `no-var` - this will impact legacy code but can be disabled on a project basis
  * `quote-props` - this is a personal preference, feel it makes code more readable and we generally do it anyway
  * `strict` - strict mode is implied in es modules so we don't need to have `"use strict"`

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,68 +1,79 @@
...
+            allowSingleLine: true
         }],
-        'camelcase': ['error', { 'properties': 'never' }],
+        camelcase: ['error', { properties: 'never' }],
         'comma-spacing': ['error', {
-            'before': false, 'after': true
+            before: false, after: true
         }],
         'comma-style': ['error', 'last'],
+        curly: ['error', 'all'],
```
Comment (AlbertoALopez) [inline index.js]: Only one I'm not sure about is this one. If its abused yes it can be messy but sometimes one liners are nice.

File: index.js [hunk 2]
```diff
@@ -1,68 +1,79 @@
...
-
+    parserOptions: {
+        ecmaVersion: 9,
+        sourceType: 'module',
+    },
     plugins: ['jasmine'],
-
-    // linting rules settings
+    // set this as the base `.eslint` file
+    root: true,
```
Comment (earobinson) [inline index.js]: is this true if we extend?

File: index.js (hunk 1 above)
Comment (earobinson) [reply index.js]: personally I love it

File: index.js [hunk 3]
```diff
@@ -1,68 +1,79 @@
...
-        'indent': ['error', 4, {
-            'MemberExpression': 0,
-            'SwitchCase': 1,
-            'VariableDeclarator': 1,
-            'outerIIFEBody': 1,
-            'FunctionDeclaration': {
-                'parameters': 1,
-                'body': 1
+        'global-require': 'error',
+        indent: ['error', 4, {
```
Comment (earobinson) [inline index.js]: 2?

Comment (agreenbloom) [reply index.js]: 2

Comment (AlbertoALopez) [reply index.js]: Frontend does 2, backend does 4?

Comment (mshelley-hubba) [reply index.js]: I think 2 is good to be consistent.

File: index.js (hunk 1 above)
Comment (iain8) [reply index.js]: they are error prone, and require more effort to refactor

Comment (AlbertoALopez) [reply index.js]: 👍

File: index.js (hunk 3 above)
Comment (drewjag) [reply index.js]: 2!

File: index.js [hunk 4]
```diff
@@ -1,68 +1,79 @@
...
+                parameters: 1,
+                body: 1
             },
+            MemberExpression: 0,
+            outerIIFEBody: 1,
+            SwitchCase: 1,
+            VariableDeclarator: 1
         }],
+        'jasmine/no-disabled-tests': 2,
+        'jasmine/no-focused-tests': 2,
```
Comment (drewjag) [inline index.js]: 👍

File: index.js (hunk 2 above)
Comment (iain8) [reply index.js]: hmm not sure this should be here, no

File: index.js (hunk 3 above)
Comment (iain8) [reply index.js]: happy to go with 2, but will set it as a "warn" initially so that it doesn't break all the builds

**Pull request merged**